main.py

```diff
@@ -1,22 +1,66 @@
-import win32com.client
+from dataclasses import dataclass
+import win32com.client as win32
+import sys
+
+if len(sys.argv) < 2:
+    print("Should be called with the name of the aspen file")
+    exit(1)
 
 # Connect to Aspen Plus
-aspen = win32com.client.Dispatch("Apwn.Document")
-aspen.InitFromFile(r"C:\Programming\delft\Aspen\cstr-ch4.apw")
-aspen.Visible = True
+aspen = win32.gencache.EnsureDispatch("Apwn.Document")
+aspen.InitFromArchive2(sys.argv[1])
+aspen.Visible = False
+aspen.SuppressDialogs = True  # Suppress windows dialogs
+
+@dataclass
+class SearchBlock:
+    data: list[tuple[str, str]]
+    children: list[str]
+
+search = {
+    "Hierarchy": SearchBlock([], ["Blocks"]),
+    "Compr": SearchBlock([("WNET", "Net Power")], []),
+    "MCompr": SearchBlock([("WNET", "Net Power")], []),
+    "Cyclone": SearchBlock([], []),
+    "Sep": SearchBlock([], []),
+    "HeatX": SearchBlock([], []),
+    "Dupl": SearchBlock([], []),
+    "Flash2": SearchBlock([], []),
+    "Heater": SearchBlock([], []),
+    "Mixer": SearchBlock([], []),
+    "Sep2": SearchBlock([], []),
+    "RPlug": SearchBlock([], []),
+    "Valve": SearchBlock([], []),
+    "RStoic": SearchBlock([], []),
+}
+
+
+data = {}
+
+def get_all_children(node):
+    return (node.Elements.Item(i) for i in range(node.Elements.Count))
 
+RECORD_TYPE = 6

```

Comment:
what is record type 6?

Reply:
This is the attribute that corresponds to record type (I got this from the aspen plus variable explorer)